databases/week2/part_four.sql

```diff
@@ -0,0 +1,57 @@
+DROP DATABASE IF EXISTS `hw_lesson2_part4`;
+
+CREATE DATABASE IF NOT EXISTS `hw_lesson2_part4`;
+
+USE `hw_lesson2_part4`;
+
+CREATE TABLE employee_details( 
+	id INT NOT NULL PRIMARY KEY, 
+    salary INT NOT NULL,
+    department_id INT UNIQUE NOT NULL, 
+    start_date DATE NOT NULL,
+    end_date DATE NULL,
+    status_id INT UNIQUE NOT NULL,
+    INDEX(start_date, end_date) 
+)ENGINE=INNODB DEFAULT CHARSET=UTF8MB4 COLLATE=UTF8MB4_UNICODE_CI;
+
+CREATE TABLE employees( 
+	id INT NOT NULL AUTO_INCREMENT PRIMARY KEY, 
+    name VARCHAR(255) NOT NULL, 
+    email VARCHAR(255) UNIQUE NOT NULL,
+    phone VARCHAR(255) NULL, 
+    CONSTRAINT fk_emp_id FOREIGN KEY(id) REFERENCES employee_details(id),
+    INDEX(name)
+)ENGINE=INNODB DEFAULT CHARSET=UTF8MB4 COLLATE=UTF8MB4_UNICODE_CI;
+
+CREATE TABLE employee_status( 
+	status_id INT NOT NULL PRIMARY KEY, 
+    name VARCHAR(255) NOT NULL, 
+    CONSTRAINT fk_emp_status_id FOREIGN KEY(status_id) REFERENCES employee_details(status_id)
+)ENGINE=INNODB DEFAULT CHARSET=UTF8MB4 COLLATE=UTF8MB4_UNICODE_CI; 
+
+CREATE TABLE shops( -- parent table 
+	shop_id INT NOT NULL PRIMARY KEY, 
+    name VARCHAR(255) NOT NULL UNIQUE, 
+    phone VARCHAR(255) NULL
+)ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci; 
```

Comment:
Why do you want table shop as your design seems to be about an employee & department he/she is working in...?
**PLEASE note** that table names should not be plural like `departments`=`department`,

Reply:
Hi Elias, Thank you so much for your review!
I created shop table because in hw I need to have at least many to many relationship and my shop table and employee is many to many relationship 
1 employee can work more than 1 shop and 1 shop ca have more than one employee 

if you dont think it makes sense with all relationships, let me know so I can try to find the new many to many relationship :)

Reply:
> Hi Elias, Thank you so much for your review! I created shop table because in hw I need to have at least many to many relationship and my shop table and employee is many to many relationship 1 employee can work more than 1 shop and 1 shop ca have more than one employee
> 
> if you dont think it doesnt makes sense with all relationships, let me know so I can try to find the new many to many relationship :)

HI Elias, 
I just wanted to update you that I have redesigned the whole database with hotel booking, please considering review it again and let me know if I have to adjust something else or I can merge it :)